SalesforceCommerceCloud/pwa-kit
# [POC]Create pwa-kit-config to be shared among packages

# Description

Currently, we have `pwa-kit-dev` requires `pwa-kit-runtime` as devDependency. It works in the context of pwa kit project development because both packages are installed as dependencies. However, when we execute the cli outside of that context (e.g using npx), we ran into an issue as below:
![image](https://user-images.githubusercontent.com/52219283/166622603-1d09d564-3e3a-4c1e-9024-a450f9fcaeb3.png)

The reason it failed is that npx does not install `devDependencies`. In other words, if we want to fix that, we have to declare `pwa-kit-runtime` as a dependency so npx know and install the package to execute the command. 

However, installing the whole package `pwa-kit-runtime` might not be the best idea since the bin only requires `getConfig()` to work

So I am trying to explore the idea that what if npx only needs to install `getConfig` instead of the whole `pwa-kit-runtime` at the time it wants to execute a command. Hence, this PR was created. 

Some points we need to consider:
- First of all, does it worth the effort to look into this idea since we know for the fact that `pwa-kit-dev` always needs `PWA-kit-runtime` as a dependency (regardless dependecies or devDependencies). 
- If we want to explore the idea, `getConfig` currently is written in the context of runtime. It needs to read know Managed Runtime, so if we break it down into a different package, we might want to write it agnostically. I haven't digged into this yet since I did not spend much time on this PR

Also another idea, `getConfig` only get called if the push command is used. Could we do a dynamic import here so when we want to run other commands through npx, it wont be affected? 🤔 



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes


# How to Test-Drive This PR

- `npx lerna clean`
- `npm ci` at root
- Run Retail react app, and verify everything is still working 


# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
